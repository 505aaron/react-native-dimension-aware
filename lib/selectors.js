export const getScreen = state => state.screen;
export const getScreenWidth = state => state.screen.width;
export const getScreenHeight = state => state.screen.height;

export const getWindow = state => state.window;
export const getWindowWidth = state => state.window.width;
export const getWindowHeight = state => state.window.height;
