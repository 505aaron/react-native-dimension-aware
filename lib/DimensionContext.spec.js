import React from "react";
import { View, Dimensions, Text } from "react-native";
import TestRenderer from "react-test-renderer";
import DimensionConsumer, { DimensionProvider } from "./DimensionContext";
import { getWindowWidth, getWindowHeight } from "./selectors";

describe("DimensionContext", () => {
  let component;

  beforeEach(() => {
    component = (
      <DimensionProvider>
        <DimensionConsumer>
          {dimensions => (
            <View>
              <Text>Width {getWindowWidth(dimensions)}</Text>
              <Text>Height {getWindowHeight(dimensions)}</Text>
            </View>
          )}
        </DimensionConsumer>
      </DimensionProvider>
    );
  });

  describe("lifecyle", () => {
    test("should re-render the component on updates", () => {
      const testRenderer = TestRenderer.create(component);

      Dimensions.set({ window: { width: 4000, height: 2000 } });

      testRenderer.update(component);

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    test("should remove listeners", () => {
      const testRenderer = TestRenderer.create(component);

      testRenderer.unmount();

      Dimensions.set({ window: { width: 4000, height: 2000 } });

      expect(testRenderer.toJSON()).toBeNull();
    });
  });
});
