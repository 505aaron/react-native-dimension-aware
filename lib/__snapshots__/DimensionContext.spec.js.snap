// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DimensionContext lifecyle render context 1`] = `
<View>
  <Text>
    Width 
    200
  </Text>
  <Text>
    Height 
    200
  </Text>
</View>
`;

exports[`DimensionContext lifecyle should re-render the component on updates 1`] = `
<View>
  <Text>
    Width 
    4000
  </Text>
  <Text>
    Height 
    2000
  </Text>
</View>
`;
